src/gridInterceptor/interceptorP5.js

```diff
@@ -1,68 +1,68 @@
-        gridInterceptor.populateObject(x, arguments, gridInterceptor.drawObject, details, true);
-        gridInterceptor.isCleared = false;
+            if (frameCount == 0) { // for setup
+                table.innerHTML = '';
+                details.innerHTML = '';
+                summary.innerHTML = '';
+                gridInterceptor.createShadowDOMElement(document);
+                gridInterceptor.setupObject =
+                    gridInterceptor.populateObject(x, arguments, gridInterceptor.setupObject, details, false);
+                gridInterceptor.populateObjectDetails(gridInterceptor.setupObject, gridInterceptor.drawObject, summary, details);
+                gridInterceptor.populateTable(details, gridInterceptor.setupObject);
+            } else if (frameCount == 1 || frameCount % 20 == 0) {
+                gridInterceptor.drawObject =
+                    gridInterceptor.populateObject(x, arguments, gridInterceptor.drawObject, details, true);
+                gridInterceptor.isCleared = false;
 
-        //clean the cells
-        var cells = document.getElementsByClassName('gridOutput-cell-content');
-        cells = [].slice.call(cells);
-        cells.forEach(function(cell){
-          cell.innerHTML = '';
-        });
+                //clean the cells
+                let cells = document.getElementsByClassName('gridOutput-cell-content');
+                cells = [].slice.call(cells);
+                cells.forEach(function(cell) {
+                    cell.innerHTML = '';
+                });
 
-        //concat the new objects and populate the grid
-        //TODO : make this more efficient so that it happens only ONCE per frame count
-        programObjects = gridInterceptor.setupObject.objectArray.concat(gridInterceptor.drawObject.objectArray);
-        gridInterceptor.populateObjectDetails(gridInterceptor.setupObject, gridInterceptor.drawObject, summary, details);
-        gridInterceptor.populateTable(programObjects,document);
-      }
-      if (x.name == 'redraw') { // reset some of the variables
-        gridInterceptor.drawObject = gridInterceptor.clearVariables(gridInterceptor.drawObject);
-      }
-      return originalFunc.apply(this, arguments);
-    };
-  });
-}
+                //concat the new objects and populate the grid
+                //TODO : make this more efficient so that it happens only ONCE per frame count
+                programObjects = gridInterceptor.setupObject.objectArray.concat(gridInterceptor.drawObject.objectArray);
+                gridInterceptor.populateObjectDetails(gridInterceptor.setupObject, gridInterceptor.drawObject, summary, details);
+                gridInterceptor.populateTable(programObjects, document);
+            }
+            if (x.name == 'redraw') { // reset some of the letiables
+                gridInterceptor.drawObject = gridInterceptor.clearletiables(gridInterceptor.drawObject);
+            }
+            return originalFunc.apply(this, arguments);
```

Comment:
Throwing an error here. Try changing the arguments name.

Reply:
@lm-n I don't think this a variable that we can just change. This is actually using the arguments array. This should not throw an error. Can you show me the error that you are getting?